modules/bringauto_process/SignalHandler.go

```diff
@@ -0,0 +1,70 @@
+// Functions for global signal handling
+//
+// The Register() must be called to start handling signal by this module. Then AddHandler() can be
+// called to add handlers to execute when the registered signal is received. The method returns
+// function which should be deferred after AddHandler() call by caller. This returned function
+// basically removes handler from execution after signal is received. When registered signal is
+// received, all added handlers (and not removed yet) will be executed in reverse order and then
+// the program exits with status code 1.
+//
+// Note: Do not use with concurrent programming. Can behave unexpectedly!
+
+package bringauto_process
+
+import (
+	"bringauto/modules/bringauto_log"
+	"fmt"
+	"os"
+	"sync"
+	"os/signal"
+)
+
+
+var lock sync.Mutex
+var handlers []func() error
+
+// Registers handling of specified signals to bringauto_process package
+func RegisterSignal(sig ...os.Signal) {
```

Comment:
Doesn't make sense for me. The purpose of the function is registering a signal to be handled by bringauto_process module. When `SignalHandlerRegister` would be used, i would think that the function will register Handler to be executed by this module when signal is received. Maybe `RegisterSignalHandling` is more descriptive? Or do you think add `SignalHandler` at the beginning to every function name so it is clear that these functions are one part of module to handle signals? Then i would suggest `SignalHandlerRegisterSignal`, which makes sense for me.

Reply:
The reason why I want to rename it is because the `RegisterSignal` and `AddHandler` are part of one logic component.
Now, If I read the code I am quite confused if they are part of the same logic component. The function renames can explicitly tell - ouu these functions are bound together and consist of one logic unit...

Reply:
and yes `SignalHadler` is considered as a prefix. `SignalHandlerRegisterSignal` and `SignalHandlerAddHandler` are completely ok.

Reply:
Ok, done.